### ISSUE TITLE
TypeScript template for fastify-openapi-glue cli?

### ISSUE THREAD

**Issue by hongkongkiwi** [seriousme/fastify-openapi-glue#581]
Hi there,

Thanks for the great project! It's great to be able to make the OpenAPI specs the source of truth.

Is it possible to add a typescript template to the fastify-openapi-glue generator?

**Comment by seriousme** [comment]
Hi,

last time I checked (few years ago) there was no stable way to generate Typescript types from OpenApi types.
The generator itself allows for extra templating (see https://github.com/seriousme/fastify-openapi-glue/tree/master/lib/templates). If you have ideas on how to template the types  (without resorting to too much magic)  then I'm happy to look into that.

There are plans to add Type annotations to Javascript (https://github.com/tc39/proposal-type-annotations), so I was hoping to piggyback on that one.

Kind regards,
Hans

**Comment by jcx0** [comment]
Hello, +1 thanks for this project. I also like the schema-first approach.

A while ago I was able to hack this project https://github.com/hey-api/openapi-ts to generate types for the routes. I've being using it with great success it so far for my personal project. https://github.com/jcx0/fastify-ts if anyone want to take a look, I've included an example server. Seems to support 2.0, 3.0, and 3.1 specifications, but I've only used it for 3.1. Wildcard status codes (e.g. 4XX) don't work, but I'm sure it wouldn't be too hard to implement.
